sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/state/StateStoreRDD.scala

```diff
@@ -76,7 +76,7 @@ class StateStoreRDD[T: ClassTag, U: ClassTag](
 
     store = StateStore.get(
       storeProviderId, keySchema, valueSchema, indexOrdinal, storeVersion,
-      storeConf, hadoopConfBroadcast.value.value)
+      storeConf, hadoopConfBroadcast.value.value, readOnly)
     val inputIter = dataRDD.iterator(partition, ctxt)
     storeUpdateFunction(store, inputIter)
```

Comment:
For read-only, do we still need to call `storeUpdateFunction`?

Reply:
Despite of the name, the function is used to R/W against StateStore, so still need to call for read-only case as well. (For sure, both should provide different functions.)

As I commented earlier, code divergence should help to make it be clearer. I'll look into it.